src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -608,14 +608,28 @@ void enforceMemberNotNull(SyntaxNode? syntaxOpt, LocalState state)
                         // https://github.com/dotnet/roslyn/issues/46718: give diagnostics on return points, not constructor signature
                         var exitLocation = method.DeclaringSyntaxReferences.IsEmpty ? null : method.Locations.FirstOrDefault();
                         bool constructorEnforcesRequiredMembers = method.ShouldCheckRequiredMembers();
+
+                        // Required properties can be attributed MemberNotNull, indicating that the property is set, the field will be set as well.
+                        // If we're enforcing required members (ie, the constructor is not attributed with SetsRequiredMembers), we also want to
+                        // not warn for members named in such attributes.
+                        var membersWithStateEnforcedByRequiredMembers = constructorEnforcesRequiredMembers
+                            ? method.ContainingType.GetMembersUnordered().SelectManyAsArray(
+                                predicate: member => member is PropertySymbol { IsRequired: true },
+                                selector: member =>
+                                {
+                                    var property = ((PropertySymbol)member);
+                                    return property.SetMethod?.NotNullMembers ?? property.NotNullMembers;
```

Comment:
>return property.SetMethod?.NotNullMembers ?? property.NotNullMembers; [](http://example.com/codeflow?start=36&length=69)

I am curious why we are not digging into NotNullMembers for these members as well.

Reply:
See above.